services/database-analyzer.js

```diff
@@ -204,27 +204,41 @@ function DatabaseAnalyzer(databaseConnection, config, allowWarning) {
     const schema = {};
 
     return databaseConnection.collections()
-      .then(collections => P.each(collections, async (item) => {
-        const collection = item.s;
-        const collectionName = collection && collection.namespace
-          && collection.namespace.collection;
-
-        // NOTICE: Defensive programming
-        if (!collectionName) { return; }
-
-        // NOTICE: Ignore system collections.
-        if (collectionName.startsWith('system.')) { return; }
-
-        const analysis = await analyzeMongoCollection(collectionName);
-        schema[collectionName] = {
-          fields: analysis,
-          references: [],
-          primaryKeys: ['_id'],
-          options: {
-            timestamps: isUnderscored(analysis),
-          },
-        };
-      }))
+      .then(async (collections) => {
+        if (collections.length === 0) {
+          logger.error(
+            'Your database is empty.',
+            'Please, create some collections before running generate command.',
+          );
+          await eventSender.notifyError('database_empty', 'Your database is empty.', {
+            orm: 'mongoose',
+            dialect: 'mongodb',
+          });
+          return process.exit(1);
+        }
```

Comment:
I just added these lines in this file, the rest is unchanged ;)

Reply:
@VincentMolinie @larcin if we consider the DRY principle here, this code cannot be accepted.
The error handling logic is shared with SQL database "onboarding" so there should not be code duplication.

@VincentMolinie we cannot basic copy paste in the code base anymore. You **must** consider this for any contributions.
@larcin this is smelly code, as this kind of logic was already handled for SQL databases.
A basic search for "Your database is empty" would have shown the code duplication.